ping.py

```diff
@@ -0,0 +1,26 @@
+import os
+import time
+
+from sqlalchemy import create_engine
+from sqlalchemy.engine.url import make_url
+
+from CTFd.config import Config
+
+url = make_url(Config.DATABASE_URL)
+if not url.drivername.startswith("mysql"):
+    exit(0)
```

Comment:
Shouldn't this script work regardless of database driver? Can this just be removed?

Reply:
`engine.raw_connection()` returns a underlying connection from another driver, `pymysql` in this case. `ping` is a mariadb/mysql only operation.

Reply:
With the strategy below you'd probably just want to exit if the driver is sqlite.